idaes/models_extra/gas_solid_contactors/unit_models/fixed_bed_1D.py

```diff
@@ -0,0 +1,2047 @@
+        else:
+            has_heat_of_reaction_gas_phase = False
+
+        # Create a unit model length domain
+        self.length_domain = ContinuousSet(
+            bounds=(0.0, 1.0),
+            initialize=self.config.length_domain_set,
+            doc="Normalized length domain",
+        )
+
+        self.bed_height = Var(
+            domain=Reals,
+            initialize=1,
+            doc="Bed length",
+            units=units_meta_solid("length"),
+        )
+
+        # =========================================================================
+        """ Build Control volume 1D for gas phase and
+            populate gas control volume"""
+
+        self.gas_phase = ControlVolume1DBlock(
+            transformation_method=self.config.transformation_method,
+            transformation_scheme=self.config.transformation_scheme,
+            finite_elements=self.config.finite_elements,
+            collocation_points=self.config.collocation_points,
+            dynamic=True,  # Fixed beds must be dynamic
+            has_holdup=True,  # holdup must be True for fixed beds
+            area_definition=DistributedVars.variant,
+            property_package=gas_phase.property_package,
+            property_package_args=gas_phase.property_package_args,
+            reaction_package=gas_phase.reaction_package,
+            reaction_package_args=gas_phase.reaction_package_args,
+        )
+
+        self.gas_phase.add_geometry(
+            length_domain=self.length_domain,
+            length_domain_set=self.config.length_domain_set,
+            length_var=self.bed_height,
+            flow_direction=FlowDirection.forward,
+        )
+
+        self.gas_phase.add_state_blocks(
+            information_flow=FlowDirection.forward, has_phase_equilibrium=False
+        )
+
+        if gas_phase.reaction_package is not None:
+            self.gas_phase.add_reaction_blocks(
+                has_equilibrium=gas_phase.has_equilibrium_reactions
+            )
+
+        self.gas_phase.add_material_balances(
+            balance_type=self.config.material_balance_type,
+            has_phase_equilibrium=False,
+            has_mass_transfer=has_mass_transfer_gas_phase,
+            has_rate_reactions=has_rate_reaction_gas_phase,
+        )
+
+        self.gas_phase.add_energy_balances(
+            balance_type=self.config.energy_balance_type,
+            has_heat_transfer=has_heat_transfer,
+            has_heat_of_reaction=has_heat_of_reaction_gas_phase,
+        )
+
+        self.gas_phase.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            has_pressure_change=self.config.has_pressure_change,
+        )
+
+        # =========================================================================
+        """ Build Control volume 1D for solid phase and
+            populate solid control volume"""
+
+        # Build Solid Phase StateBlock
+        # As there is no solid flow, only need a single state block
```

Comment:
This statement is confusion - as written it implies to me a single (unindexed) state block rather than a set of indexed state blocks. I would suggest like:

# As there is no solid flow, there is no need for a control volume as a set of indexed state blocks will be sufficient